Implemented front end browser-based UI tests #733

fgiroud opened the pull request:
This PR introduces Cypress as a browsed-based testing framework. Implements #733 
It includes
* The setup for Cypress
* Developer documentation
* Tests that cover basic but broad OpenRefine interface

wetneb reviewed (commented):
Thanks a lot, it looks great to me and it is fantastic to see the documentation being updated in the same pull request.

fgiroud commented:
Hi, thx !
I have added the entry in the sidebar, passed the markdown to autoformat, and linked the installation from the build-test-run page.
Also added a proper license to the package.json

Regarding the build-test-run, the command will be replaced by a simple call to `./refine ui_test`, but for this I need to setup Travis first, otherwise it will break the build. Coming soon.

Best,
- Florian

wetneb reviewed (approved):
Thanks again. I am happy to merge this in this current state.

As discussed in the last testing meeting, the tests currently rely on the "production" instance of OpenRefine (the one running on the standard port 3333, which opens by default on the "official" workspace for the machine).  For `./refine ui_test`, I would expect that the script starts an instance of OpenRefine on a different port (say 3334 for instance), and on a temporary workspace directory, so that the test workspace remains isolated from any other projects the developer might have on their machine.

Currently, running the tests on my machine give 4 failures (with Chromium, Linux), but I am sure we will get occasions to debug them down the line, when we set them up in Travis.
